modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/atomic/GridNearAtomicUpdateFuture.java

```diff
@@ -690,10 +700,20 @@ private void sendUpdateRequests(Map<UUID, PrimaryRequestState> mappings) {
         if (locUpdate != null) {
             cache.updateAllAsyncInternal(cctx.localNode(), locUpdate,
                 (req, res) -> {
-                    if (syncMode != FULL_ASYNC)
-                        onPrimaryResponse(res.nodeId(), res, false);
-                    else if (res.remapTopologyVersion() != null)
-                        ((GridDhtAtomicCache<?, ?>)cctx.cache()).remapToNewPrimary(req);
+                    CacheOperationContext prevOpCtx = cctx.operationContextPerCall();
```

Comment:
Here and in GridNearAtomicAbstractUpdateFuture we use same pattern to deal with context.
Let's introduce `UpdateReplyClosureContextAware` which will contains all the logic related to operation context set/clear.